### ISSUE TITLE
ManagedKubernetesClient configuration policy should be REQUIRE

### ISSUE THREAD

**Issue by mattrpav** [fabric8io/kubernetes-client#3700]
### Describe the bug

ManagedKubernetesClient does not handle activate( ) method being called with null properties. The Configuration policy should be set to 'REQUIRE' in order to only activate when a config is present.

### Fabric8 Kubernetes Client version

5.5.0

### Steps to reproduce

1. Deploy kubernetes-client in osgi w/o a config file
2. Observe NPE

```
Caused by: java.lang.NullPointerException
	at io.fabric8.kubernetes.client.osgi.ManagedKubernetesClient.activate(ManagedKubernetesClient.java:143) ~[kubernetes-client-5.3.1-bundle.jar:?]
	... 73 more
```

### Expected behavior

Do not try to activate a ManagedKubernetesClient without a configuration file

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.20.12

### Environment

macOS

**Comment by rohanKanojia** [comment]
@mattrpav : I have a question in the context of https://github.com/fabric8io/kubernetes-client/issues/3864 . Do you have some reproducer project which I can try out and reproduce for ManagedOpenShiftClient?

**Comment by rohanKanojia** [comment]
@mattrpav : I wanted to ask another question, Have you upgraded to Fabric8 Kubernetes Client 6.x version and checked whether this fix is working for you?

I'm trying to prepare a reproducer for https://github.com/fabric8io/kubernetes-client/issues/3864 but at the moment I'm getting this exception with ManagedKubernetesClient:
```
llPointerException - null]
[INFO] k8s: java.lang.NullPointerException: null
[INFO] k8s: 	at io.fabric8.kubernetes.client.osgi.NamespacedKubernetesClientAdapter.getClient(NamespacedKubernetesClientAdapter.java:120) ~[!/:?]
[INFO] k8s: 	at io.fabric8.kubernetes.client.osgi.NamespacedKubernetesClientAdapter.resources(NamespacedKubernetesClientAdapter.java:131) ~[!/:?]
```

I'm not sure whether I'm using `ManagedKubernetesClient` correctly or not. Could you please review [my reproducer](https://github.com/rohankanojia-forks/kubernetes-client-osgi-demo/blob/main/src/main/java/org/eclipse/jkube/quickstart/karaf/K8sBookGenerator.java#L42) and provide some feedback ?

**Comment by mattrpav** [comment]
@rohanKanojia I don't use OpenShift, so not sure how it'll behave.

I'm happy to test out 6.x-- it'll be a bit later this week. I've got 3 other oss fixes in the queue.

**Comment by rohanKanojia** [comment]
@mattrpav : You don't have to worry about OpenShift. I will fix that as part of #3864 

My question is regarding your OSGI project setup in ManagedKubernetesClient (any rough code snippet would be super helpful). I can replicate that onto OpenShift. Could you please review my reproducer to see if's trying to reproduce correctly?

**Comment by rohanKanojia** [comment]
@mattrpav : polite ping, Did you manage to find time to look into this? It would be really helpful if you could provide a simple reproducer .

The [reproducer](https://github.com/rohankanojia-forks/kubernetes-client-osgi-demo) I prepared doesn't seem to be calling `ManagedKubernetesClient.activate()` method which results in NPE
